sycl/doc/extensions/experimental/sycl_ext_oneapi_bf16_math.asciidoc

```diff
@@ -0,0 +1,105 @@
+# Bfloat16 math functions extension for DPC++: = SYCL_ONEAPI_bf16_math
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+:dpcpp: pass:[DPC++]
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+
+:blank: pass:[ +]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+
+== Notice
+
+Copyright (c) 2021-2021 Intel Corporation.  All rights reserved.
+
+IMPORTANT: This specification is a draft.
+
+NOTE: The APIs described in this specification are experimental. Future versions of this extension may change these APIs in ways that are incompatible with the versions described here.
+
+NOTE: Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are
+trademarks of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc.
+used by permission by Khronos.
+
+This extension is written against the SYCL 2020 revision 4 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+## Contributors
+
+* Jack Kirk
+
+## Introduction
+
+This document proposes extending the `fma`, `fmin`, `fmax` and `fabs` SYCL floating point math functions to support the `bfloat16` type introduced in the `SYCL_EXT_INTEL_BF16_CONVERSION` extension. This proposal assumes that devices which support the `SYCL_EXT_INTEL_BF16_CONVERSION` extension have the `bfloat16` scalar data type: `bfloat16`, and the `bfloat16` vector data types: `bfloat16_1`, `bfloat16_2`, `bfloat16_3`, `bfloat16_4`, `bfloat16_8` and `bfloat16_16` available at compile-time, in line with corresponding `half` types that are available at compile time on devices that
```

Comment:
I was told that I should initially propose the support of equivalent vector types to those that are currently supported in SYCL 2020 for the half case: this would mean that e.g. `bfloat16_2` is a vector of two `bfloat16`.  If we continue with the plan of implementing the corresponding vector types and these bfloat16 vector types are not defined in another extension then I imagine that we would define the vector types in this extension; but as you say it probably makes sense to have a single `bfloat16` extension. I think at this point we mainly need to know whether Intel also plans to support these functions at all with `bfloat16` in their backends or whether this extension would only apply to the CUDA backend at this moment in time? Further on from this if there is any idea on whether Intel would like to support bfloat16 vector types for these functions that would also be welcome.

I have asked whether the `aspect::ext_intel_bf16_conversion` could be implemented as a backend agnostic aspect here: https://github.com/intel/llvm/pull/5393#issuecomment-1042810593